io.openems.edge.controller.pvinverter.reversepowerrelay/src/io/openems/edge/controller/pvinverter/reversepowerrelay/ReversePowerRelayImpl.java

```diff
@@ -0,0 +1,169 @@
+package io.openems.edge.controller.pvinverter.reversepowerrelay;
+
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.types.ChannelAddress;
+import io.openems.edge.common.component.AbstractOpenemsComponent;
+import io.openems.edge.common.component.ComponentManager;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.controller.api.Controller;
+import io.openems.edge.pvinverter.api.ManagedSymmetricPvInverter;
+import io.openems.edge.common.channel.BooleanReadChannel;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Controller.PvInverter.ReversePowerRelay", //
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE //
+)
+public class ReversePowerRelayImpl extends AbstractOpenemsComponent
+		implements ReversePowerRelay, Controller, OpenemsComponent {
+
+	private final Logger log = LoggerFactory.getLogger(ReversePowerRelayImpl.class);
+
+	@Reference
+	private ComponentManager componentManager;
+
+	private String pvInverterId;
+
+	private int powerLimit30Percent = 0;
+	private int powerLimit60Percent = 0;
+
+	private ChannelAddress inputChannelAddress0Percent = null;
+	private ChannelAddress inputChannelAddress30Percent = null;
+	private ChannelAddress inputChannelAddress60Percent = null;
+	private ChannelAddress inputChannelAddress100Percent = null;
+
+	private ManagedSymmetricPvInverter pvInverter;
+
+	public ReversePowerRelayImpl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				Controller.ChannelId.values(), //
+				ReversePowerRelay.ChannelId.values() //
+		);
+	}
+
+	@Activate
+	private void activate(ComponentContext context, Config config) {
+		super.activate(context, config.id(), config.alias(), config.enabled());
+		this.pvInverterId = config.pvInverter_id();
+
+		this.powerLimit30Percent = config.powerLimit30();
```

Comment:
Couldn't this be calculated based on the Inverters Power?

Like maybe add one Config field to add the max Power of the Inverter and based on this Limit it can calculate how much is 30% or 60% off of it ? :)

Reply:
Thought about that, too. My intention was to have a little more "freedom" how much the inverter gets limited. But you´re right. It´s a bit more handy to calculate the values.